src/main/java/com/automationexercise/Pages/P02_SearchProductPage.java

```diff
@@ -45,7 +42,7 @@ public void PressSearchButton() {
     //Verify
 
     public Boolean VerifyUserNavigatedToAllProductsPage() {
-        explicitlyWaitForVisibility(driver, (By) AllProductsPage);
+        explicitlyWaitForVisibility(driver, AllProductPage);
         return AllProductsPage.isDisplayed();
```

Comment:
There is a util function to wait on the element and there returns if displayed or not, you can use it instead of re-implement it.

Reply:
I can't find it, Can you mention it's name.

Reply:
```
  public static boolean verifyElementVisible(By by) {
        try {
            explicitlyWaitForVisibility(getDriver(), by);
            LogUtils.info("Verify element visible " + by);
            return true;
        } catch (Exception e) {
            LogUtils.error(e.getMessage());
            return false;
        }
    }
``` 
@AhmedMohmd1